modules/mbedtls/Kconfig

```diff
@@ -111,6 +112,37 @@ config MBEDTLS_DEBUG_LEVEL
 	  This makes Zephyr call mbedtls_debug_set_threshold() function during
 	  mbedTLS initialization, with the configured debug log level.
 
+choice MBEDTLS_DEBUG_EXTRACT_BASENAME
+	prompt "Extract basename from filenames"
+	default MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME
+
+config MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME
+	bool "At buildtime"
+	help
+	  Adds compile options, which should convert full source paths in
+	  __FILE__ macro to files' basenames. This will reduce code footprint
+	  when debug messages are enabled.
+
+	  This is compiler dependent, so if it does not work then please
+	  fallback to MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_RUNTIME instead.
```

Comment:
This is very unfortunate.

Someone building a project with a 3rd party toolchain will not see this message.
Such a user will just see (at best) a compilation error, but with no knowledge of why / how to fix.

We should not default to something which may not work generally.
That makes it much harder for users to know what to change, as well as require them to change configuration for any sample that uses mbed TLS.

Please ensure this setting is only default selection for toolchains that are known to work. (gcc)

Reply:
>Such a user will just see (at best) a compilation error, but with no knowledge of why / how to fix.

Well, build error will be triggered for all Zephyr applications, as there is already a similar `zephyr_cc_option` in Zephyr's root `CMakeLists.txt`:

https://github.com/zephyrproject-rtos/zephyr/blob/2f5fef960e2310f5e4dd0bc1b9cd5fe9c5a73581/CMakeLists.txt#L401-L405

What I meant by:
```
	  This is compiler dependent, so if it does not work then please
	  fallback to MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_RUNTIME instead.
```
is that it might not actually trim file paths. I assumed that this will not trigger build error, because we already do similar trick in root `CMakeLists.txt`.

> We should not default to something which may not work generally.

Agree on that, but what about root `CMakeLists.txt`?

> Please ensure this setting is only default selection for toolchains that are known to work. (gcc)

Any hint on how to achieve that?

Reply:
@marc-hb Could you provide any input regarding `-fmacro-prefix-map` usage in this particular context? At least you are the author of 28a5657f1ff6 ("cmake: zephyr_cc_option(-fmacro-prefix-map=${ZEPHYR_BASE}=.)").

Reply:
> > We should not default to something which may not work generally.
> 
> Agree on that, but what about root `CMakeLists.txt`?

There is a small difference to the CMakeLists.txt code and that is that the `zephyr_cc_option()` function will go through a compiler test, and if the compiler doesn't support the flag it will be discarded.

For example, it's not until [llvm 10](https://releases.llvm.org/10.0.0/tools/clang/docs/ReleaseNotes.html#new-compiler-flags) that this flag was supported.
So custom compilers based on older llvm (or gcc <v8.0.0) will not support the flag.
But those flags are not visible as a user config option.

When you expose the setting in Kconfig, then it gives the impression that this will work, thus misleading the user.
In fact the flag might be discarded on some compilers and such situations this can lead to confusion.
Why is my flag not being applied when enabled in Kconfig ?

> 
> > Please ensure this setting is only default selection for toolchains that are known to work. (gcc)
> 
> Any hint on how to achieve that?

Generally one can test for the compiler in Kconfig, for example like this:
```
config MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME
    ....
    depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
```

because we know this toolchain definitely supports this setting.

Reply:
Thanks for detailed explanation! I decided to enable "buildtime" only for zephyr toolchain, but still leave the option to select it manually for 3rdparty toolchains:
```
choice MBEDTLS_DEBUG_EXTRACT_BASENAME
	prompt "Extract basename from filenames"
	default MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
	default MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_RUNTIME
```